pkg/core/helper_test.go

```diff
@@ -142,6 +146,116 @@ func newDumbBlock() *block.Block {
 	}
 }
 
+func TestCreateLongChain(t *testing.T) {
+	// This test takes rather long time and exists solely to create dumps.
+	t.Skip()
+	const saveChain = true
+	const outFile = "../../chain.acc" // root of git repo
+	// Takes 1 minute for 100 tx per block and 5000 blocks.
+	const blockAmount = 5000
+	const txPerBlock = 100
+	const contract = `
+	package contract
+	import "github.com/nspcc-dev/neo-go/pkg/interop/storage"
+	var ctx = storage.GetContext()
+	func Put(key, value []byte) {
+		storage.Put(ctx, key, value)
+	}`
+
+	acc, err := wallet.NewAccount()
+	require.NoError(t, err)
+	h := acc.Contract.ScriptHash()
+
+	bc := newTestChain(t)
+	defer bc.Close()
+	bc.config.VerifyBlocks = false
+
+	txMoveNeo := bc.transferFundsFromOwner(t, bc.contracts.NEO.Hash, h, native.NEOTotalSupply, 0, 2)
+	txMoveGas := bc.transferFundsFromOwner(t, bc.contracts.GAS.Hash, h, 2_000_000_000_000_000, 0, 2)
+	require.NoError(t, bc.AddBlock(bc.newBlock(txMoveNeo, txMoveGas)))
+
+	tx, avm := newDeployTxFromReader(t, "TEST", strings.NewReader(contract))
+	tx.Signers = []transaction.Signer{{
+		Account: h,
+		Scopes:  transaction.CalledByEntry,
+	}}
+	tx.NetworkFee = 10_000_000
+	tx.ValidUntilBlock = bc.BlockHeight() + 1
+	require.NoError(t, acc.SignTx(tx))
+	require.NoError(t, bc.AddBlock(bc.newBlock(tx)))
+
+	contractHash := hash.Hash160(avm)
+	key := make([]byte, 10)
+	value := make([]byte, 10)
+	nonce := uint32(0)
+	for i := bc.BlockHeight(); i < blockAmount; i++ {
+		txs := make([]*transaction.Transaction, txPerBlock)
+		for j := 0; j < txPerBlock; j++ {
+			nonce++
+			rand.Read(key)
+			rand.Read(value)
+
+			w := io.NewBufBinWriter()
+			emit.AppCallWithOperationAndArgs(w.BinWriter, contractHash, "put", key, value)
+			require.NoError(t, w.Err)
+			tx := transaction.New(netmode.UnitTestNet, w.Bytes(), 4_000_000)
+			tx.ValidUntilBlock = i + 1
+			tx.NetworkFee = 4_000_000
+			tx.Nonce = nonce
+			tx.Signers = []transaction.Signer{{
+				Account: h,
+				Scopes:  transaction.CalledByEntry,
+			}}
+			require.NoError(t, acc.SignTx(tx))
+			txs[j] = tx
+		}
+
+		require.NoError(t, bc.AddBlock(bc.newBlock(txs...)))
+	}
+
+	if saveChain {
+		dumpChain(t, bc, outFile)
+	}
+}
+
+func (bc *Blockchain) transferFundsFromOwner(t *testing.T, contractHash, to util.Uint160, amount int64,
+	nonce, validUntil uint32) *transaction.Transaction {
+	// Move some NEO to one simple account.
+	tx := newNEP5Transfer(contractHash, neoOwner, to, amount)
+	tx.ValidUntilBlock = validUntil
+	tx.Nonce = nonce
+	tx.Signers = []transaction.Signer{{
+		Account:          neoOwner,
+		Scopes:           transaction.CalledByEntry,
+		AllowedContracts: nil,
+		AllowedGroups:    nil,
+	}}
+	require.NoError(t, signTx(bc, tx))
+	return tx
+}
+
+func dumpChain(t *testing.T, bc *Blockchain, path string) {
```

Comment:
We have a very similar piece of code in cli, perhaps we can make a separate helper with this method? It also would be helpful for neo-bench.

Reply:
May be something like `Blockchain.DumpTo(filename string)`.

Reply:
An when combined with `RestoreFrom` this can be useful for tests/benchmarks.